server/shared/publishing/publishing.service.js

```diff
@@ -15,7 +16,8 @@ class PublishingService {
   }
 
   publishActivity(activity) {
-    return this.queue.add(() => publishActivity(activity));
+    return this.queue.add(() => publishActivity(activity)
+      .then(data => oauthClient.send(data) && data));
```

Comment:
Sorry for nitpicking, but would this be a bit easier to read if we'd move callback with the promise chain to a variable?
For example
```suggestion
    const publish = () => publishActivity(activity)
      .then(data => oauthClient.send(data) && data);
    return this.queue.add(publish);
```